developer/frameworks/articles/user-associated-data/04-filtering-and-searching.markdown

```diff
@@ -0,0 +1,76 @@
+---
+header-id: filtering-and-searching-uad-marked-entities
+---
+
+# Filtering and Searching UAD-Marked Entities
+
+[TOC levels=1-4]
+
+When you 
+
+![Figure 1: The UAD UI has filtering and searching elements you can leverage in
+custom entities.](../../images/uad-filter-search.png)
+
+To add filtering and searching for your custom entities, implement three
+methods in the `UADDisplay` class (in your application's `-uad` module):
+
+## Filtering
+
+The `isSiteScoped` method returns a boolean based on whether or not the
+entities can be associated with a particular site. This is used to determine
+which filter they will be associated with ("instance", "personal-site", or
+"regular-sites").
+
+<!-- I want to show some realistic dummy code for third party devs. Is this fine?-->
+<!--Straight out of BaseModelUADDisaply: -->
+
+```java
+@Override
+public boolean isSiteScoped() {
+    Class<FooEntry> typeClass = FooEntry.class;
+
+    if (GroupedModel.class.isAssignableFrom(typeClass)) {
+        return true;
+    }
+
+    return false;
+}
+```
```

Comment:
I personally wouldn't recommend this particular code example since it's designed to be a catch-all for most cases, but also is designed to be overridden.  A better example would be to simply return `true` or `false` (which is probably going to be the most common case in a custom implementation).

Reply:
Okay, I'll update the method description to read

> The `isSiteScoped` method returns a boolean based on whether or not the entities can be associated with a particular site: return `false` if not, and `true` if the entities are scoped to a site. This is used to determine which filter they will be associated with ("instance", "personal-site", or "regular-sites").

And the code example to just be 

```java
@Override
public boolean isSiteScoped() {

    return false;
}
```

:+1:  or :-1:  ? cc @drew